### PR TITLE
Look for model type in dependencies if its not found in the project directly.

### LINKED ISSUE #251: Scaffolding fails if model class is in a dependency (project/ library) of the project on which scaffolding is being run.
Project A has a dependeny on Project B. 
Project B has model class 
If you try to run scaffolding on Project A by using model class from Project B, it fails with the below error: 
`No model type returned for type: <class name>`

image

Related #249 #140

### PR DESCRIPTION
**Pull request by prafullbhosale**
Fixes #251

cc @abpiskunov

### CODE REVIEW

**Comment by barrytang** [comment]
TFS Bug#<a href="https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_workItems#id=233431&fullScreen=false&_a=edit">233431</a>

File: src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/EntityFrameworkServices.cs [hunk 1]
```diff
@@ -289,6 +290,36 @@ public class EntityFrameworkServices : IEntityFrameworkService
             return true;
         }
 
+        // Look for the model type in the current project. 
+        // If its not found in the current project, look in the dependencies.
+        private Type GetModelTypeFromAssembly(string modelTypeName, Assembly assembly)
+        {
+            Type modelType = assembly.GetType(modelTypeName);
```
Comment (abpiskunov) [inline src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/EntityFrameworkServices.cs]: GetType can throw several kinds of exceptions, do we protect from it?

File: src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/EntityFrameworkServices.cs [hunk 2]
```diff
@@ -289,6 +290,36 @@ public class EntityFrameworkServices : IEntityFrameworkService
...
+            if (modelType == null)
+            {
+                // Need to look in the dependencies of this project now.
+                var dependencies = assembly.GetReferencedAssemblies().GetEnumerator();
```
Comment (abpiskunov) [inline src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/EntityFrameworkServices.cs]: I don't remember , but does build happens before we do that? If not assume user added a new model in the project B and did not build yet, then tries to scaffold in project A, which would find Project B assembly, but no new type yet... What is the plan for this case?

Comment (prafullbhosale) [inline src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/EntityFrameworkServices.cs]: At this point a build has already happened. 
Also, before reaching here, we already make sure that the modelType actually exists using the ModelTypesLocator. 

In this particular part, we are trying to get the modelType from the in-memory compilation of the project (and its dependencies) after the DbContext has been added/ modified.

**Comment by abpiskunov** [comment]
:shipit:

**Pull request merged**